python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -15,15 +16,23 @@
 
 from dagster_components.core.schema.base import ResolvableModel
 from dagster_components.core.schema.metadata import ResolvableFieldInfo
-from dagster_components.core.schema.resolver import TemplatedValueResolver
+from dagster_components.core.schema.resolver import ResolveContext
 
 
-class OpSpecModel(ResolvableModel["OpSpecModel"]):
+class OpSpecModel(ResolvableModel):
     name: Optional[str] = None
     tags: Optional[dict[str, str]] = None
 
-    def resolve(self, resolver: TemplatedValueResolver) -> "OpSpecModel":
-        return OpSpecModel(**resolver.resolve_obj(self.model_dump(exclude_unset=True)))
+    def resolve(self, context: ResolveContext) -> "OpSpecModel":
```

Comment:
can this not be eliminated with the shenanigans?

Reply:
Because it's resolving to itself, things don't work out as nicely -- you need to use string reference for the generic type when defining `OpSpecModel(ResolvableModel["OpSpecModel"])`, which doesn't work when we're reading the generic type at runtime

Reply:
Another good reason to move this out of this class hierarchy.